analytical_engine/core/grape_instance.cc

```diff
@@ -1097,8 +1121,24 @@ bl::result<std::shared_ptr<DispatchResult>> GrapeInstance::OnReceive(
     break;
   }
   case rpc::RUN_APP: {
-    BOOST_LEAF_AUTO(context_key, query(params, cmd->query_args));
-    r->set_data(context_key);
+    // For vertex context and arrowprojected fragment we return a new graph.
+    BOOST_LEAF_AUTO(graph_name, params.Get<std::string>(rpc::GRAPH_NAME));
+    BOOST_LEAF_AUTO(frag_wrapper,
+                    object_manager_.GetObject<IFragmentWrapper>(graph_name));
+    BOOST_LEAF_AUTO(return_graph, params.Get<bool>(rpc::RETURN_GRAPH));
+    std::string context_key = "ctx_" + generateId();
+    BOOST_LEAF_AUTO(context_result,
+                    query(params, cmd->query_args, context_key, frag_wrapper));
```

Comment:
Looks incorrect as the auto-assign `context_result` is never use.

Reply:
I suggest don't touch the implementation of `query()` and use the returned `context_key` instead.